.github/workflows/code-formatting.yml

```
@@ -7,6 +7,21 @@ on:  # Trigger the workflow on push or pull request, but only for the master bra
     branches: [master, "release/*"]
 
 jobs:
+  imports-check-isort:
+    name: Check valid import formatting with isort
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install isort
+        run: pip install isort==5.6.4
```

do we need to specify a specific version here? Should it install the latest one when a new version is be released in future.
```suggestion
        run: pip install isort
```

---

From experience, I prefer to specify the version explicitly, and to keep it the same as the one for the pre-commit hook (that needs to be specified in `.pre-commit-config.yaml`). For example, it happened to me once that a new release of black broke compatibility and the CI checks started to fail, while the pre-commit checks (with a fixed version) were still all clear.

By specifying the version, we eliminate the risk of silent version changes breaking the CI. When we update the version manually, we could always check if any breaking changes require us to update the code formatting.

---

yeah, valid point :+1: